x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/action/PostAnalyticsEventAction.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.analytics.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.application.analytics.event.AnalyticsEvent;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class PostAnalyticsEventAction extends ActionType<PostAnalyticsEventAction.Response> {
+
+    public static final PostAnalyticsEventAction INSTANCE = new PostAnalyticsEventAction();
+
+    public static final String NAME = "cluster:admin/xpack/application/analytics/post_event";
+
+    private PostAnalyticsEventAction() {
+        super(NAME, Response::readFromStreamInput);
+    }
+
+    public static class Request extends ActionRequest implements AnalyticsEvent.Context {
+
+        private final String eventCollectionName;
+
+        private final String eventType;
+
+        private final long eventTime;
+
+        private final boolean debug;
+
+        private final BytesReference payload;
+
+        private final XContentType xContentType;
+
+        public Request(String eventCollectionName, String eventType, boolean debug, XContentType xContentType, BytesReference payload) {
+            this(eventCollectionName, eventType, debug, System.currentTimeMillis(), xContentType, payload);
+        }
+
+        public Request(
+            String eventCollectionName,
+            String eventType,
+            boolean debug,
+            long eventTime,
+            XContentType xContentType,
+            BytesReference payload
+        ) {
+            this.eventCollectionName = eventCollectionName;
+            this.eventType = eventType;
+            this.debug = debug;
```

Comment:
@kderusso I kept `debug` instead of suggestions like `verbose`.

Indeed it is intended to be used with the debug flag of the JS tag that will output event processed by the server into the dev console of the browser (feature that I plan to implement later).